dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -112,30 +135,46 @@ public static void main(String[] arguments)
    */
   public static ImmutableList<Path> artifactsToClasspath(List<Artifact> artifacts)
       throws RepositoryException {
+    
+    ListMultimap<Path, DependencyPath> multimap = artifactsToPaths(artifacts);
+    LinkedHashSet<Path> set = new LinkedHashSet<>(); // to remove duplicate keys
+    for (Entry<Path, DependencyPath> entry : multimap.entries()) {
+      set.add(entry.getKey());
+    }
+    
+    return ImmutableList.<Path>builder().addAll(set).build();
+  }
+  
+  
+  // Multimap is a pain, maybe just use LinkedHashMap<Path, List<DependencyPath>>
```

Comment:
Because it's ~8 different types with unclear behavior around whether key and/or value order is or is not preserved that I can never remember without checking the Javadocs and can't get right without really solid test coverage. At the end of the day, it's just too complex.

Reply:
The `Immutable*` types always preserve insertion order. But I guess it's up to you.

Reply:
That's a surprise. To me "Immutable" means you can't add or remove from the collection. It doesn't suggest to me that an unordered type like set has a predictable iteration order, much less that it preserves insertion order. It feels more natural to specify a type that is ordered by definition, i.e. a list, when we want to preserve order.